### PR TITLE
Add BarcodeFormat to result

### PR DESCRIPTION
**Pull request by tvh**
This currently supports all types present on android. For iOs, only a subset is supported.

This should fix https://github.com/juliuscanute/qr_code_scanner/issues/63.

### CODE REVIEW

**Comment by dspicher** [comment]
@tvh thanks for this great PR! Would you mind adapting the example to show both the barcode type and barcode contents?

**Comment by dspicher** [comment]
@tvh Can you squash the commits into one please? I'm happy to do it myself if you check the appropriate checkbox.

**Comment by tvh** [comment]
Done

**Comment by dspicher** [comment]
Great, thanks!

**Comment by dspicher** [comment]
@tvh so sorry, I still didn't get around to testing this. It's still on my radar, will get it done soon!

**Comment by juliansteenbakker** [comment]
@tvh can you merge the upstream changes and fix the merge conflicts? After that ill merge this and make it available for 0.2.0.

**Comment by tvh** [comment]
Done

File: example/lib/main.dart [hunk 1]
```diff
@@ -18,14 +19,21 @@ class QRViewExample extends StatefulWidget {
...
+  Barcode result;
   var flashState = flashOn;
   var cameraState = frontCamera;
   QRViewController controller;
   final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
 
   @override
   Widget build(BuildContext context) {
+    String resultText;
+    if (result == null) {
```
Comment (juliansteenbakker) [inline example/lib/main.dart:31]: I think it's better to do this check inline with a ternary operator.

**Pull request merged**